esp-hal/src/gpio/mod.rs

```diff
@@ -307,7 +308,7 @@ pub trait RtcPinWithResistors: RtcPin {
 }
 
 /// Common trait implemented by pins
-pub trait Pin: Sealed {
+pub trait Pin: Sealed + Unpin {
```

Comment:
This is technically a breaking change btw, I don't think it needs a migration guide but could you note it in the log at least

Reply:
How is this breaking?

Reply:
Hmm, maybe not actually - I was thinking about someone using `Pin` in some where bounds, but I suppose that wouldn't make a difference actually. Ignore me :D